packages/headless/src/controllers/commerce/facets/core/utils/headless-core-commerce-facet-utils.test.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  buildMockCommerceCategoryFacetValue,
+  buildMockCommerceDateFacetValue,
+  buildMockCommerceNumericFacetValue,
+  buildMockCommerceRegularFacetValue,
+} from '../../../../../test/mock-commerce-facet-value';
+import {FacetValueState} from '../../../../core/facets/facet/headless-core-facet';
+import {FacetType} from '../headless-core-commerce-facet';
+import {findIfHasActiveValues} from './headless-core-commerce-facet-utils';
+
+describe('#findIfHasActiveValues', () => {
+  describe('when facet type is "category', () => {
+    const type = 'hierarchical' as FacetType;
+    describe('when current value has ancestors', () => {
+      const values = [
+        buildMockCommerceCategoryFacetValue({
+          value: 'A',
+          path: ['A'],
+          children: [
+            buildMockCommerceCategoryFacetValue({
+              value: 'B',
+              path: ['A', 'B'],
+              children: [
+                buildMockCommerceCategoryFacetValue({
+                  value: 'C',
+                  path: ['A', 'B', 'C'],
+                }),
+                buildMockCommerceCategoryFacetValue({
+                  value: 'D',
+                  path: ['A', 'B', 'D'],
+                  state: 'selected',
+                }),
+                buildMockCommerceCategoryFacetValue({
+                  value: 'E',
+                  path: ['A', 'B', 'E'],
+                }),
+              ],
+            }),
+          ],
+        }),
+      ];
+      it('returns "true"', () => {
+        expect(findIfHasActiveValues(values, type)).toBe(true);
+      });
```

Comment:
I don't think this granularity is necessary (we could probably merge the `describe` with the `it`), but I don't mind leaving the test definition as-is.

Reply:
Agreed, I was testing this structure to see if I liked it and I'm not sure I do either 😁